docs/google-analytics.md

```diff
@@ -0,0 +1,78 @@
+## how google analytics works
+
+Since angular is a single page application, I implemented router checking. So when a user activates the Router function, it is recorded as visiting a new part of the page.
+
+For optimal user tracking and continued work on Treecreate.dk after we have launched, it is important that we work eventEmitters into the more important actions of our users. ex:
+
+1. how long they spend creating a design
+2. how many designs they create
+3. how go from one page to the other
+   etc etc.
+
+When you are working on an important action for the page like: sign in, create account etc, please add a google event emitter, so we can track usage.
+
+## how to use the EVENT EMITTER for google analytics
+
+I have created a google-analytics.service.ts file that you can use to trigger events. We should only use this for events we believe can bring us valuable insight from our users. Here is an example of an add_to_cart event:
+
+```ts
+import { Component, OnInit } from '@angular/core';
+import { GoogleAnalyticsService } from '../google-analytics/google-analytics.service';
+
+@Component({
+  selector: 'google-button-test',
+  templateUrl: './google-button-test.component.html',
+  styleUrls: ['./google-button-test.component.css'],
+})
+export class GoogleButtonTestComponent implements OnInit {
+  constructor(public googleAnalyticsService: GoogleAnalyticsService) {}
+
+  ngOnInit(): void {}
+
+  testGoogleEventEmitter() {
+    console.log('testing google event emitter');
+    this.googleAnalyticsService.eventEmitter(
+      'add_to_cart',
+      'shop',
+      'cart',
+      'click',
+      1
+    );
+  }
+}
+```
+
+As you can see, you need to do the following:
+
+1. import GoogleAnalyticsService
+2. add it in the constroctor -> public googleAnalyticsService: GoogleAnalyticsService
+3. add a line of code specifying what kind of event it is, that should trigger when the event happens.
+   -> this.googleAnalyticsService.eventEmitter( "add_to_cart", "shop", "cart", "click", 1 );
+
+## How to write the correct event for .eventEmitter()
+
+googleAnalyticsService.eventEmitter( "eventName", "eventCategory", "eventAction", "eventLabel", "eventValue" );
```

Comment:
Place the code in a codeblock

Also, provide a bit more information on what that code is and where it should be put. Remember that this documentation is also meant for developers that have not touched google analytics before and don't know where to put the code.
```suggestion
`googleAnalyticsService.eventEmitter( "eventName", "eventCategory", "eventAction", "eventLabel", "eventValue" );`
```